### PR TITLE
[besu] Decopule Vault CURD Operations in helm charts 

### PR DESCRIPTION
**Pull request by ravikiranjanjanam**
Changes:

Vault CURD operations have been decoupled and are moved to a script where the vault Script will be created as a configmap in each organizational namespace and will be mounted as a volume in helm charts.

Reviewers:
  @suvajit-sarkar @jagpreetsinghsasan @sownak 

Issues:
#2221

### CODE REVIEW

File: platforms/hyperledger-besu/charts/orion_key_mgmt/templates/job.yaml [hunk 1]
```diff
@@ -136,3 +122,7 @@ spec:
                 fi
                 COUNTER=`expr "$COUNTER" + 1`
               fi
+          volumeMounts:
+            - name: scripts-volume
+              mountPath: /scripts/bevel-vault.sh
+              subPath: bevel-vault.sh
```
Comment (suvajit-sarkar) [inline platforms/hyperledger-besu/charts/orion_key_mgmt/templates/job.yaml:128]: add new line

File: platforms/shared/charts/vault-script/scripts/bevel-vault.sh [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                  -H "X-Vault-Token: ${VAULT_TOKEN}" \
+                  -H "Content-Type: application/json" \
+                  -X POST \
+                  -d @${2} \
+                  ${VAULT_ADDR}/v1/${1})
+}
+
+##############################################################################################################################
+################################################## AWS Secret Manager functions starts #######################################
+##############################################################################################################################
```
Comment (suvajit-sarkar) [inline platforms/shared/charts/vault-script/scripts/bevel-vault.sh:64]: Is this tested ? Remove if not. Use of aws secret manager and akv would need changes in ansible side aswell

**Review by suvajit-sarkar — CHANGES_REQUESTED** [review]

**Review by sownak — CHANGES_REQUESTED** [review]
generally looks good. But, as we now have multiple vaults, I would expect a if condition in all the  Besu charts on which vault to be used.

File: platforms/shared/charts/vault-script/scripts/bevel-vault.sh [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+                                    --data-raw '{"value": "'"$JSON_PAYLOAD"'"}' )
+}
+
+
+##############################################################################################################################
+################################################## Vault main handler function ###############################################
+##############################################################################################################################
+
+vaultBevelFunc() {
+    if [[ $VAULT_TYPE = "hashicorp" ]]; then
```
Comment (suvajit-sarkar) [inline platforms/shared/charts/vault-script/scripts/bevel-vault.sh:123]: @sownak the vault conditions are determined here based on the vault type passed into container env variable.

File: platforms/hyperledger-besu/charts/orion_key_mgmt/templates/job.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ravikiranjanjanam) [reply platforms/hyperledger-besu/charts/orion_key_mgmt/templates/job.yaml]: New line added

File: platforms/shared/charts/vault-script/scripts/bevel-vault.sh (hunk 2 above)
Comment (ravikiranjanjanam) [reply platforms/shared/charts/vault-script/scripts/bevel-vault.sh]: Removed the code for aws and Azure secret manager as this is not tested

**Comment by ravikiranjanjanam** [comment]
> generally looks good. But, as we now have multiple vaults, I would expect a if condition in all the Besu charts on which vault to be used.

Removed the code for other vaults like aws secret manager and azure as this is not tested, we are passing Vault_type as environmental variable within the helm chart consuming from helm releases Vaules ,default vault is set to hashicorp and  'if condition' was placed within the shell script. If we are expanding to other vaults, we could have a new field 'type'  in network.yaml under vault section

File: platforms/hyperledger-besu/charts/generate_ambassador_certs/templates/job.yaml [hunk 4]
```diff
@@ -67,57 +71,35 @@ spec:
...
           mkdir -p ${OUTPUT_PATH}
 
-          vault_secret_key=${VAULT_SECRET_ROOTCA}
+          vault_secret_key="${VAULT_SECRET_ENGINE}/${VAULT_SECRET_ROOTCA}"
           echo "Checking certs in ${vault_secret_key}"
-          LOOKUP_SECRET_RESPONSE=$(curl -sS --header "X-Vault-Token: ${VAULT_CLIENT_TOKEN}" ${VAULT_ADDR}/v1/${vault_secret_key} | jq -r 'if .errors then . else . end')
-          CA_PEM=$(echo ${LOOKUP_SECRET_RESPONSE} | jq -r '.data.data["rootca_pem"]' 2>&1)
-          CA_KEY=$(echo ${LOOKUP_SECRET_RESPONSE} | jq -r '.data.data["rootca_key"]' 2>&1)
+          vaultBevelFunc "readJson" ${vault_secret_key} 
+          CA_PEM=$(echo ${VAULT_SECRET} | jq -r '.data["rootca_pem"]' 2>&1)
```
Comment (sownak) [inline platforms/hyperledger-besu/charts/generate_ambassador_certs/templates/job.yaml:94]: which version of vault is this tested on?

**Review by sownak — CHANGES_REQUESTED** [review]

File: platforms/hyperledger-besu/charts/generate_ambassador_certs/templates/job.yaml (hunk 4 above)
Comment (ravikiranjanjanam) [reply platforms/hyperledger-besu/charts/generate_ambassador_certs/templates/job.yaml:94]: Its tested on 1.7.1

**Review by suvajit-sarkar — APPROVED** [review]
LGTM, tested ok. New issue will be create to incorporate condition of for vault

File: platforms/hyperledger-besu/charts/generate_ambassador_certs/templates/job.yaml (hunk 4 above)
Comment (suvajit-sarkar) [reply platforms/hyperledger-besu/charts/generate_ambassador_certs/templates/job.yaml:94]: the data.data is required, but its already sending the .data from the script
VAULT_SECRET=$(curl --header "X-Vault-Token: ${VAULT_TOKEN}" ${VAULT_ADDR}/v1/${1} | jq -r 'if .errors then "null" else .data end')

**Review by sownak — APPROVED** [review]

**Pull request merged**